src/vs/code/electron-main/app.ts

```diff
@@ -153,6 +153,90 @@ export class CodeApplication extends Disposable {
 
 		//#endregion
 
+		//#region Request filtering
+
+		// Block all SVG requests from unsupported origins
+		const supportedSvgSchemes = new Set([Schemas.file, Schemas.vscodeFileResource, Schemas.vscodeRemoteResource, 'devtools']);
+
+		// But allow them if the are made from inside an webview
+		const isSafeFrame = (requestFrame: WebFrameMain | undefined): boolean => {
+			for (let frame: WebFrameMain | null | undefined = requestFrame; frame; frame = frame.parent) {
+				if (frame.url.startsWith(`${Schemas.vscodeWebview}://`)) {
+					return true;
+				}
+			}
+			return false;
+		};
+
+		const isSvgRequestFromSafeContext = (details: Electron.OnBeforeRequestListenerDetails | Electron.OnHeadersReceivedListenerDetails): boolean => {
+			return details.resourceType === 'xhr' || isSafeFrame(details.frame);
+		};
+
+		const isAllowedVsCodeFileRequest = (details: Electron.OnBeforeRequestListenerDetails) => {
+			const frame = details.frame;
+			if (!frame || !this.windowsMainService) {
```

Comment:
How does this condition accommodate resource requests from the web worker extension host ?

Reply:
In my testing, the web worker iframe has the same processId as the main window. I believe this is because they are from the same origin?

Do you think this is a safe assumption? If not, I could also allow requests from iframes that are themselves served from `vscode-file` urls

Reply:
> In my testing, the web worker iframe has the same processId as the main window. I believe this is because they are from the same origin?

Oh okay then it should be fine.